llvm/lib/Target/ARM/MCTargetDesc/ARMAsmBackend.cpp

```diff
@@ -579,13 +579,24 @@ unsigned ARMAsmBackend::adjustFixupValue(const MCAssembler &Asm,
   case ARM::fixup_arm_uncondbl:
   case ARM::fixup_arm_condbl:
   case ARM::fixup_arm_blx:
+    // Check that the fixup value is legal.
+    Value = Value - 8;
+    if (!isInt<25>(Value)) {
```

Comment:
Should this be `isInt<26>`? The ARM B/BL/BLX instructions take a 2's complement 24-bit immediate, left-shifted by 2.

Reply:
You're right. I mixed up the check between the ASM value and the encoded one.